File: Duran_TuranMert.dart
void main() {
  var phoneList = {  //INITIALIZATION 
	  "Ahmet": 053245934321,
	  "Ali": 05424456454,
      "Turan" : 05834323133
  };
  print(phoneList['Ahmet']); // GETTIN VALUE OF GIVEN KEY
  phoneList["Eklemeddin"] = 052319324156; // ADDING ELEMENT
  phoneList.remove("Ahmet");//removing Element
  phoneList["Eklemeddin"] = 042315534332; // MODIFYING ELEMENT
  print(phoneList.containsKey("Turan")); // SEARCH KEY
  print(phoneList.containsValue(05834323133)); // SEARCH VALUE
  foo(phoneList);
  
}
void foo( var phoneList) { 
  phoneList.forEach((k, v) {
    print('{ key: $k, value: $v }');
  });
}


// https://programming-idioms.org/idiom/8/initialize-a-new-map-associative-array/202/dart ALL initialization of ass. arrays in all languages
// https://dartpad.dev ONLINE COMPILER OF DART
// https://bezkoder.com/dart-map/ ALL FUNCTIONS OF DART MAP 

File: Duran_TuranMert.lua
local phoneList = { ["Ahmet"] = 5454, ["Turan"] = 2523}
print(phoneList["Ahmet"]) -- print value with key
phoneList["Halil"] = 3414 -- add
phoneList["Ahmet"] = nil -- remove
phoneList["Halil"] = 3423  -- modify
if phoneList["Halil"] then print("Yes it includes") end -- Search by key
    for i,v in pairs(phoneList) do
   print(i)
   print(v)
    end

    function get_key_for_value( t, value ) -- Search by value method
        for k,v in pairs(t) do
          if v==value then return k end
        end
        return nil
      end
    print(get_key_for_value(phoneList, 3423))
    
foo = function ()
    for i,v in pairs(phoneList) do
    io.write("Key : ")
   print(i)
    io.write("Value : ")
   print(v)
    end
  end


foo() -- Call Function Prints all keys and values